exchanges/request/limit.go

```diff
@@ -74,11 +76,59 @@ func (r *Requester) InitiateRateLimit(ctx context.Context, e EndpointLimit) erro
 		return nil
 	}
 
-	if r.limiter != nil {
-		return r.limiter.Limit(ctx, e)
+	if r.limiter == nil {
+		return nil
```

Comment:
I think this should return err. Weird to call "initial rate limit" when its unset. Add protection surrounding the caller in `doRequest`